src/adagrams.js

```diff
@@ -1,15 +1,126 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+  const letterPool={
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1
+  };
+  const availableLetters = [];
+    
+  for (const [letter,frequency] of Object.entries(letterPool)){
+    for (let i = 0;i< frequency;i++){
+      availableLetters.push(letter)}   
+  }
+  const hand=[]
+  for (let i = 0; i < 10; i++) {
+    const thisLetter = availableLetters[Math.floor(Math.random() * availableLetters.length)]
+    hand.push(thisLetter)
+    let letterIndex = availableLetters.indexOf(thisLetter)
+    availableLetters.splice(letterIndex, 1)
+}
 
+return hand;
+};
+    
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const upperCase = input.toUpperCase();
+
+  for (let letter of upperCase) {
+    if (lettersInHand.includes(letter)) {
+        let index = lettersInHand.indexOf(letter)
+        lettersInHand.splice(index, 1)
+    } else {
+        return false;
+    }
+  }
+  return true;
+};
+const letterScores = {
+  A: 1,
+  B: 3,
+  C: 3,
+  D: 2,
+  E: 1,
+  F: 4,
+  G: 2,
+  H: 4,
+  I: 1,
+  J: 8,
+  K: 5,
+  L: 1,
+  M: 3,
+  N: 1,
+  O: 1,
+  P: 3,
+  Q: 10,
+  R: 1,
+  S: 1,
+  T: 1,
+  U: 1,
+  V: 4,
+  W: 4,
+  X: 8,
+  Y: 4,
+  Z: 10
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+
+  let score = 0;
+  let bonus = 8;
```

Comment:
Nice, I like that you put the extra bonus points in a variable. Since it is a constant value that we hardcode to 8 and don't change, we can rename it BONUS. 

Also, since you don't reassign `bonus`, we should use const